### PR TITLE
NetworkMessage methods

### PR DESCRIPTION
**Pull request by ranisalt**
Rebases and closes #1854

### CODE REVIEW

File: src/luascript.cpp [hunk 1]
```diff
@@ -5606,6 +5609,43 @@ int LuaScriptInterface::luaNetworkMessageReset(lua_State* L)
...
+	NetworkMessage* message = getUserdata<NetworkMessage>(L, 1);
+	if (message) {
+		lua_pushnumber(L, message->getLength());
+	} else {
+		lua_pushnil(L);
+	}
+	return 1;
+}
+
+
```
Comment (nekiro) [inline src/luascript.cpp:5648]: double new line?

File: src/networkmessage.h [hunk 2]
```diff
@@ -128,6 +128,14 @@ class NetworkMessage
 			return info.position;
 		}
 
+		bool setBufferPosition(MsgSize_t pos) {
+			if (pos < NETWORKMESSAGE_MAXSIZE - INITIAL_BUFFER_POSITION) {
+				position = pos + INITIAL_BUFFER_POSITION;
```
Comment (nekiro) [inline src/networkmessage.h:133]: info.position instead of position

File: src/networkmessage.h [hunk 3]
```diff
@@ -128,6 +128,14 @@ class NetworkMessage
 			return info.position;
 		}
 
+		bool setBufferPosition(MsgSize_t pos) {
```
Comment (nekiro) [inline src/networkmessage.h:131]: Why is MsgSize_t used here?

**Comment by MillhioreBT** [comment]
😍This one looks very interesting, because I like to use a lot of network messages 👍

**Review by djarek — APPROVED** [review]

File: src/networkmessage.h (hunk 3 above)
Comment (ranisalt) [reply src/networkmessage.h:131]: That's the type of `info.position`

**Review by nekiro — APPROVED** [review]

**Pull request merged**